subprojects/snapshots/src/main/java/org/gradle/internal/vfs/watch/impl/JdkFileWatcherRegistry.java

```diff
@@ -93,6 +95,7 @@ public void stopWatching(ChangeHandler handler) throws IOException {
             }
         } finally {
             close();
+            handler.close();
```

Comment:
Why do we need to close this now? And why here? If we get the handler injected in the constructor I'd say whoever created it should dispose of it, too.

Reply:
I am returning the statistics for logging from stopWatching. Not there is no need to close the handler any more.